kong/plugins/datadog/statsd_logger.lua

```diff
@@ -19,10 +19,16 @@ local stat_types = {
 local statsd_mt = {}
 statsd_mt.__index = statsd_mt
 
+local env_datadog_agent_host = os.getenv 'KONG_DATADOG_AGENT_HOST'
+local env_datadog_agent_port = os.getenv 'KONG_DATADOG_AGENT_PORT'
 
 function statsd_mt:new(conf)
   local sock   = udp()
-  local _, err = sock:setpeername(conf.host, conf.port)
+  if conf.use_env then
+    local _, err = sock:setpeername(env_datadog_agent_host, env_datadog_agent_port)
```

Comment:
since `err` is declared as a local here, it will always be `nil` when checking at line 32. So the `local` declaration should go before the `if/then`, similar to the local `sock`.

Reply:
An error needs to be returned if the env vars haven't been set;
```suggestion
    if not (env_datadog_agent_host or env_datadog_agent_port) then
      return nil, "both KONG_DATADOG_AGENT_HOST and KONG_DATADOG_AGENT_PORT must be set"
    end
    _, err = sock:setpeername(env_datadog_agent_host, env_datadog_agent_port)
```

Reply:
I restructured the control-flow.  I'm pretty happy with it now, let me know what you think.  I'm starting the write tests now.